public/src/components/contributionTypes.tsx

```diff
@@ -0,0 +1,293 @@
+import Switch from '@material-ui/core/Switch';
+import Radio from '@material-ui/core/Radio';
+import RadioGroup from '@material-ui/core/RadioGroup';
+
+import Button from '@material-ui/core/Button';
+import SaveIcon from '@material-ui/icons/Save';
+import RefreshIcon from '@material-ui/icons/Refresh';
+
+enum ContributionType {
+  ONE_OFF = 'ONE_OFF',
+  MONTHLY = 'MONTHLY',
+  ANNUAL = 'ANNUAL'
+}
+
+interface ContributionTypeSetting {
+  contributionType: ContributionType,
+  isDefault?: boolean
+}
+
+enum Region {
+  GBPCountries = 'GBPCountries',
+  UnitedStates = 'UnitedStates',
+  EURCountries = 'EURCountries',
+  International = 'International',
+  Canada = 'Canada',
+  AUDCountries = 'AUDCountries',
+  NZDCountries = 'NZDCountries'
+}
+
+type ContributionTypes = {
+  [r in Region]: ContributionTypeSetting[]
+}
+
+interface DataFromServer {
+  value: ContributionTypes,
+  version: string,
+}
+
+function isRegion(s: string): s is Region {
+  return Object.values(Region).includes(s)
+}
+
+function isContributionType(s: string): s is ContributionType {
+  return Object.values(ContributionType).includes(s)
+}
+
+const allContributionTypes = [
+  {contributionType: ContributionType.ONE_OFF, label: 'One-off'},
+  {contributionType: ContributionType.MONTHLY, label: 'Monthly'},
+  {contributionType: ContributionType.ANNUAL, label: 'Annual'}
+];
+
+const styles = ({ palette, spacing, mixins }: Theme) => createStyles({
+  form: {
+    display: 'flex',
+    flexDirection: 'row'
+  },
+  button: {
+    marginRight: spacing.unit * 2,
+    marginBottom: spacing.unit * 2
+  },
+  buttons: {
+    marginTop: spacing.unit * 2,
+    marginLeft: spacing.unit * 4
+  },
+  regions: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  },
+  region: {
+    paddingRight: spacing.unit * 4,
+    paddingLeft: spacing.unit * 4,
+    borderBottom: `1px solid ${palette.grey['300']}`,
+    borderRight: `5px solid ${palette.grey['300']}`,
+    marginBottom: spacing.unit * 4,
+    flexDirection: 'row'
+  },
+  contributionTypes: {
+    flexDirection: 'column',
+    display: 'flex',
+    borderRight: `1px solid ${palette.grey['300']}`,
+    marginRight: spacing.unit * 4
+  },
+  default: {
+    flexDirection: 'column',
+    display: 'flex'
+  },
+  label: {
+    marginTop: spacing.unit * 4
+  }
+});
+
+interface Props extends WithStyles<typeof styles> {}
+
+class ContributionTypesComponent extends React.Component<Props, ContributionTypes> {
+  state: ContributionTypes;
+  previousStateFromServer: DataFromServer | null;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      GBPCountries: [],
+      UnitedStates: [],
+      EURCountries: [],
+      International: [],
+      Canada: [],
+      AUDCountries: [],
+      NZDCountries: []
+    };
+    this.previousStateFromServer = null;
+  }
+
+  componentWillMount(): void {
+    this.fetchStateFromServer();
+  }
+
+  fetchStateFromServer(): void {
```

Comment:
before we add a third page we should refactor this duplication

Reply:
I've just moved the common GET/POST logic to /utils/